tests/stdlib/topenssl.nim

```diff
@@ -0,0 +1,66 @@
+
+proc rsaPublicEncrypt*(fr: string): cstring =
+  var rsa: PRSA
+  var to: cstring
+  var frLen, rsaLen: cint
+
+  var mKey = "-----BEGIN PUBLIC KEY-----\n"
+  mKey.add PUBKEY.wrapWords(64)
+  mKey.add "\n-----END PUBLIC KEY-----"
+
+  var bio = bioNew(bioSMem())
+  if BIO_write(bio, mKey.cstring, mKey.len.cint) < 0:
+    raise newException(Exception, "ERROR")
+  rsa = PEM_read_bio_RSA_PUBKEY(bio, nil, nil, nil)
+  if rsa == nil:
+    raise newException(Exception, "ERROR")
+  if BIO_free(bio) < 0:
+    raise newException(Exception, "ERROR")
+
+  frLen = fr.len.cint
+  rsaLen = RSA_size(rsa)
+  to = newString(rsaLen)
+  let frdata = cast[ptr cuchar](fr.cstring)
+  var todata = cast[ptr cuchar](to)
+  let ret = RSA_public_encrypt(frLen, frdata, todata, rsa, RSA_PKCS1_PADDING)
+  if ret == -1:
+    raise newException(Exception, "ERROR")
+  RSA_free(rsa)
+  return to
+
+proc rasPrivateDecrypt*(fr: cstring): cstring =
+  var rsa: PRSA
+  var to: cstring
+  var rsaLen: cint
+
+  var mKey = "-----BEGIN RSA PRIVATE KEY-----\n"
+  mKey.add PRIkEY.wrapWords(64)
+  mKey.add "\n-----END RSA PRIVATE KEY-----"
+
+  var bio = bioNew(bioSMem())
+  if BIO_write(bio, mKey.cstring, mKey.len.cint) < 0:
+    raise newException(Exception, "ERROR")
+  rsa = PEM_read_bio_RSAPrivateKey(bio, nil, nil, nil)
+  if rsa == nil:
+    raise newException(Exception, "ERROR")
+  if BIO_free(bio) < 0:
+    raise newException(Exception, "ERROR")
+
+  rsaLen = RSA_size(rsa)
+  to = newString(rsaLen)
+  let frdata = cast[ptr cuchar](fr)
+  var todata = cast[ptr cuchar](to)
+  let ret = RSA_private_decrypt(rsaLen, frdata, todata, rsa, RSA_PKCS1_PADDING)
+  if ret == -1:
+    raise newException(Exception, "ERROR")
+  RSA_free(rsa)
+  return to
+
+var res = "TEST"
+var miwen = rsaPublicEncrypt(res)
+var mingwen = rasPrivateDecrypt(miwen)

```

Comment:
`doAssert mingwen == res` ? or something else? otherwise the test doesn't test as much